scripts/Microsoft.DotNet.Cli.Build.Framework/BuildContext.cs

```diff
@@ -42,7 +42,7 @@ public BuildTargetResult RunTarget(string name, bool force)
             BuildTarget target;
             if (!Targets.TryGetValue(name, out target))
             {
-                Reporter.Verbose.WriteLine($"Skipping undefined target: {name}");
+                throw new Exception($"Undefined target: {name}");
```

Comment:
cc @anurse not throwing or returning here was causing a null ref.

I threw this in to help with debuggability. Any reason why we wouldn't want this?